ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/StaticQueryExecutor.java

```diff
@@ -465,32 +505,73 @@ private static ComparisonTarget extractWhereClauseTarget(final ComparisonExpress
     }
   }
 
-  private static void validateSelects(final Select select) {
-    final List<SelectItem> selectItems = select.getSelectItems();
+  private static boolean isSelectStar(final List<SelectItem> selects) {
+    return selects.size() == 1 && selects.get(0) instanceof AllColumns;
+  }
 
-    if (selectItems.size() != 1
-        || selectItems.get(0) instanceof SingleColumn
-    ) {
-      throw new KsqlException("Static queries currently only support a 'SELECT *' projections");
+  private static Result handleSelects(
+      final Result input,
+      final ConfiguredStatement<Query> statement,
+      final KsqlExecutionContext executionContext,
+      final Analysis analysis
+  ) {
+    final List<SelectItem> selectItems = statement.getStatement().getSelect().getSelectItems();
+    if (input.rows.isEmpty() || isSelectStar(selectItems)) {
+      return input;
+    }
+
+    final LogicalSchema.Builder schema = LogicalSchema.builder();
+    schema.keyColumns(input.schema.key());
+
+    final ExpressionTypeManager expressionTypeManager = new ExpressionTypeManager(
+        input.schema,
+        executionContext.getMetaStore()
+    );
+
+    final List<SelectExpression> selects = new ArrayList<>(selectItems.size());
+    for (int idx = 0; idx < analysis.getSelectExpressions().size(); idx++) {
```

Comment:
For simple loop variables, are three letter names really necessary? i might be better

Reply:
lol. super nit!  I prefer `idx` and we've no style guide that says I can't ;)